api/src/main/java/brooklyn/basic/AbstractBrooklynObjectSpec.java

```diff
@@ -121,4 +121,21 @@ protected final void checkIsImplementation(Class<?> val, Class<? super T> requir
         if (Modifier.isAbstract(val.getModifiers())) throw new IllegalStateException("Implementation "+val+" is abstract, but must be a non-abstract class");
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        if (obj==null) return false;
```

Comment:
i don't think it breaks the contract of hashcode -- if a spec is modified in such a way that the hashcode changes, then the equality semantics will have changed in the expected way.

as we already have `Object.equals/hashcode` i think this implementation is an improvement, and putting things into a test convenience would just make things harder to find.

making the spec immutable would be the elegant way to solve this; i wouldn't be strongly opposed although i don't see any major benefit at this time for the extra code. (is spec mutability confusion really a problem?)